legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/sdlc/LegendDependencyManagement.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finos.legend.engine.ide.lsp.extension.sdlc;
+
+import java.nio.file.Path;
+import java.util.List;
+import org.finos.legend.engine.ide.lsp.extension.features.LegendVirtualFileSystemContentInitializer;
+import org.finos.legend.engine.language.pure.grammar.to.PureGrammarComposer;
+import org.finos.legend.engine.language.pure.grammar.to.PureGrammarComposerContext;
+import org.finos.legend.engine.shared.core.api.grammar.RenderStyle;
+import org.finos.legend.sdlc.protocol.pure.v1.PureModelContextDataBuilder;
+import org.finos.legend.sdlc.serialization.EntityLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class LegendDependencyManagement implements LegendVirtualFileSystemContentInitializer
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(LegendDependencyManagement.class);
+    private final List<LegendVirtualFile> dependenciesVirtualFiles;
+
+    public LegendDependencyManagement()
+    {
+        this.dependenciesVirtualFiles = loadDependenciesVirtualFiles();
+    }
+
+    private static List<LegendVirtualFile> loadDependenciesVirtualFiles()
+    {
+        LOGGER.info("Processing dependencies from classpath");
+        try (EntityLoader loader = EntityLoader.newEntityLoader(LegendDependencyManagement.class.getClassLoader()))
+        {
+            PureModelContextDataBuilder fromClasspathBuilder = PureModelContextDataBuilder.newBuilder();
+            fromClasspathBuilder.addEntitiesIfPossible(loader.getAllEntities());
+            PureGrammarComposer pureComposer = PureGrammarComposer.newInstance(PureGrammarComposerContext.Builder.newInstance()
+                    .withRenderStyle(RenderStyle.PRETTY)
+                    .build());
+            String pureGrammar = pureComposer.renderPureModelContextData(fromClasspathBuilder.build());
```

Comment:
There some entities that we cannot generator Pure grammar for. We probably ought to handle that case, since it's going to come up. Maybe we should probably just pretty print the entity as JSON in that case?

Reply:
Hmm I rather fail in that scenario, as the json is not useful either, and cannot be parse as pure file.

Also, wont this handle scenarios where the entity cannot be converted to protocol?

`fromClasspathBuilder.addEntitiesIfPossible(loader.getAllEntities());`

Reply:
Yes, that line will filter out the entities which cannot be converted to protocol. But not all valid protocol can be converted into Pure grammar. 

And while this case isn't super common, it does occur. If you have a project with a large number of dependencies, then you're reasonably likely to encounter at least one entity which cannot be converted into grammar.

With this code as it is, if you get a single entity which cannot be converted into Pure grammar, then you won't get Pure grammar from any.

So maybe it would be better to convert entity by entity, rather than all together? Of course, if you're going to discard any entity, there's no guarantee that compilation will work.

Reply:
I change this so we do one entity at the time, and if error occurs, I append a "comment" with the error and the stack trace.  This is not ideal, but given the tools we have, this is a good 1st approach to handle failures.